Sources/MessagingPushFCM/MessagingPushFCM+PushConfigs.swift

```diff
@@ -0,0 +1,46 @@
+import FirebaseCore
+import FirebaseMessaging
+import Foundation
+#if canImport(UIKit)
+import UIKit
+#endif
+
+extension MessagingPushFCM {
+    @available(iOSApplicationExtension, unavailable)
+    func setupAutoFetchDeviceToken() {
+        swizzleDidRegisterForRemoteNotifications()
+        UIApplication.shared.registerForRemoteNotifications()
+    }
+
+    @available(iOSApplicationExtension, unavailable)
+    private func swizzleDidRegisterForRemoteNotifications() {
+        let appDelegate = UIApplication.shared.delegate
+        let appDelegateClass: AnyClass? = object_getClass(appDelegate)
+        let originalSelector = #selector(UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+        let swizzledSelector = #selector(application(_:didRegisterForRemoteNotificationsWithDeviceToken:))
+        swizzle(forOriginalClass: appDelegateClass, forSwizzledClass: MessagingPushFCM.self, original: originalSelector, new: swizzledSelector)
+    }
+
+    private func swizzle(forOriginalClass: AnyClass?, forSwizzledClass: AnyClass?, original: Selector, new: Selector) {
+        guard let swizzledMethod = class_getInstanceMethod(forSwizzledClass, new) else { return }
+        guard let originalMethod = class_getInstanceMethod(forOriginalClass, original) else {
+            // Add method if it doesn't exist
+            class_addMethod(forOriginalClass, new, method_getImplementation(swizzledMethod), method_getTypeEncoding(swizzledMethod))
+            return
+        }
+        method_exchangeImplementations(originalMethod, swizzledMethod)
+    }
+
+    // Swizzled method for APN device token.
+    // Fetch the FCM token using the Firebase delegate method when the APN token is set.
+    @objc
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        Messaging.messaging().apnsToken = deviceToken
+        Messaging.messaging().token(completion: { token, _ in
+            guard let token = token else {
+                return
+            }
+            Self.shared.registerDeviceToken(fcmToken: token)
+        })
+    }
```

Comment:
```suggestion
        // Registers listener with FCM SDK to always have the latest FCM token. Used to automatically register it with the SDK. 
        Messaging.messaging().token(completion: { token, _ in
            guard let token = token else {
                return
            }
            Self.shared.registerDeviceToken(fcmToken: token)
        })
    }
```

Added comment.

Reply:
I wonder, since this code is registering a callback function with the FCM SDK, it feels more natural to me for this block of code to exist in `MessagingPushFCM.initialize` function instead of in a swizzled function that could be called multiple times.

Reply:
Thanks for the review, Levi ! If I add this FCM callback method in `MessagingPushFCM.initialize` then it may npt work as expected as the token may be generated after initialization method.

Reply:
Added comment.

Reply:
> may not work as expected as the token may be generated after initialization method.

This is a non-blocker for this PR, but I still want to learn more about why putting `Messaging.messaging().token(completion: )` in `MessagingPushFCM.initialize` would not work. Can you provide more details on why this would not work? Maybe steps that could reproduce the behavior of it not working? 

If `Messaging.messaging().token(completion: )` will only work in certain places in the code, I think a comment should be added to this code block explaining that. So we know in the future where this block of code should go.

Reply:
This is because that the token might get generated after a delay and not immediately. One such reason is that FCM token needs APN token and it could get created later as in my case it is. 

Steps to reproduce:
- Make the change as above
- Open FCM-CocoaPods app
- Run FCM-CocoaPods app from this branch

You'll notice that when placed in initialization, the token is not generated. I can get on a call to explain this if needed.

I hope this helps.